### PR TITLE
VMware: Add disk_mode parameter in vmware_guest_disk

### PR DESCRIPTION
**Pull request by gen2kes**
Added support to select disk_mode.

##### SUMMARY
Added ability to select disk_mode between ['persistent', 'independent_persistent', 'independent_nonpersistent']
Fixes issue #58592.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
vmware_guest_disk

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_guest_disk.py [hunk 1]
```diff
@@ -90,6 +90,11 @@
      - '     - C(eagerzeroedthick) eagerzeroedthick disk'
      - '     - C(thick) thick disk'
      - '     Default: C(thick) thick disk, no eagerzero.'
+     - ' - C(disk_mode) (string): Type of disk mode. Valid values are:'
+     - '     - C(persistent) Changes are immediately and permanently written to the virtual disk. This is default.'
+     - '     - C(independent_persistent) Same as persistent, but not affected by snapshots.'
+     - '     - C(independent_nonpersistent) Changes to virtual disk are made to a redo log and discarded at power off, but not affected by snapshots.'
+     - '     Default: C(thick) thick disk, no eagerzero.'
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/vmware/vmware_guest_disk.py]:
```suggestion

```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Thanks for the contribution. Could you please add an integration test for this change ? Thanks.

**Comment by gen2kes** [comment]
Hi @Akasurde , 
I've uploaded the tests for the modified module vmware_guest_disk as requested. Please check it if you wish.
Regards

**Review by goneri — APPROVED** [review]

**Review by jillr — APPROVED** [review]
rebuild_merge

**Pull request merged**